kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/Constraints.java

```diff
@@ -448,12 +448,26 @@ public String toString() {
 
   public Map<String, String> toQueryMap() {
     Map<String, String> query = Maps.newLinkedHashMap();
+    return toQueryMap(query);
+  }
+
+  /**
+   * Get a normalized query map for the constraints. A normalized query map will
+   * be equal in value and iteration order for any logically equivalent set of
+   * constraints.
+   */
+  public Map<String, String> toNormalizedQueryMap() {
+    Map<String, String> query = Maps.newTreeMap();
+    return toQueryMap(query);
+  }
+
+  private Map<String, String> toQueryMap(Map<String, String> queryMap) {
     for (Map.Entry<String, Predicate> entry : constraints.entrySet()) {
       String name = entry.getKey();
       Schema fieldSchema = SchemaUtil.fieldSchema(schema, strategy, name);
-      query.put(name, Predicates.toString(entry.getValue(), fieldSchema));
+      queryMap.put(name, Predicates.toNormalizedString(entry.getValue(), fieldSchema));
```

Comment:
Sorry, this is my fault. I suggested using a single method, but I forgot that we don't want to change the form of the predicate for `toQueryMap()`. This should only call `toNormalizedString` when called through the normalized accessor. Either a flag passed to the private method or reversing this change would fix it.